contracts/GenArt721CoreV3.sol

```diff
@@ -155,21 +158,29 @@ contract GenArt721CoreV3 is ERC721, Ownable, IGenArt721CoreContractV3 {
         internal
         returns (uint256 _tokenId)
     {
+        // checks & effects
         uint256 nextTokenId = (_projectId * ONE_MILLION) +
             projects[_projectId].invocations;
 
-        projects[_projectId].invocations++;
+        // increment invocations, register complete timestamp if applicable
+        if (
+            ++projects[_projectId].invocations ==
+            projects[_projectId].maxInvocations
+        ) {
```

Comment:
I'm tempted to split this up for readability purposes.

Reply:
The thought here is to eliminate one SLOAD per mint to better optimize gas.
Want me to quantify with a test?

Reply:
Yeah it would be nice to quantify just to make sure we're not being overly optimizooooring here–that said, given that this is in the mint flow, my inclination is to go as optimized as possible, so the solution to my concern here may just be that we make it more explicit in the comment why we do this in one line to avoid the extra SLOAD.

Reply:
The following adds 162 gas, or $0.03/mint @100gwei, @ETH=$2k
```projects[_projectId].invocations++;
        // increment invocations, register complete timestamp if applicable
        if (
            ++projects[_projectId].invocations ==
            projects[_projectId].maxInvocations
        ) {
```
I can get even fancier and get the cost down to 34 less gas with the following, which might be overall actually easier to read?
```
function _mintToken(address _to, uint256 _projectId)
        internal
        returns (uint256 _tokenId)
    {
        // checks & effects
        // increment project's invocations, them move it to memory
        uint256 _invocationsAfter = ++projects[_projectId].invocations;
        uint256 _invocationsBefore = _invocationsAfter - 1;
        uint256 thisTokenId = (_projectId * ONE_MILLION) + _invocationsBefore;
        if (_invocationsAfter == projects[_projectId].maxInvocations) {
            _completeProject(_projectId);
        }

        bytes32 tokenHash = keccak256(
            abi.encodePacked(
                thisTokenId,
                blockhash(block.number - 1),
                randomizerContract.returnValue()
            )
        );

        tokenIdToHash[thisTokenId] = tokenHash;

        // interactions
        _mint(_to, thisTokenId);

        // Do not need to also log `projectId` in event, as the `projectId` for
        // a given token can be derived from the `tokenId` with:
        //   projectId = tokenId / 1_000_000
        emit Mint(_to, thisTokenId);

        return thisTokenId;
    }
```

Reply:
Hrm yeah my inclination is to go with the later 34-gas-less approach, as it I think is more readable indeed and also saves users money. WDYT @ryley-o ?

Reply:
agreed! pushed! 😄